packages/discount-code-generator/test/main.spec.js

```diff
@@ -1,11 +1,61 @@
-import findSum from '../src/main'
+import discountCodeGenerator from '../src/main'
 
-describe('findSum', () => {
-  test('main module is a function', () => {
-    expect(typeof findSum).toBe('function')
+describe('discountCodeGenerator', () => {
+  const options = {
+    quantity: 100,
+    length: 15,
+    prefix: 'CT',
+  }
+
+  const data = {
+    name: {
+      en: 'William',
+    },
+    description: {
+      en: 'something good',
+    },
+    cartDiscounts: [],
+    cartPredicate: 'expensive cart',
+    isActive: true,
+    maxApplications: 10,
+    maxApplicationsPerCustomer: 2,
+  }
+
+  test('should throw if no options or options is not valid', () => {
+    expect(() => discountCodeGenerator())
+      .toThrow(/The generator requires valid parameters/)
+    expect(() => discountCodeGenerator('foo'))
+      .toThrow(/The generator requires valid parameters/)
+  })
+
+  test('should throw if no data object is passed', () => {
+    expect(() => discountCodeGenerator(options))
+      .toThrow(/The generator requires discount data/)
+  })
+
+  test('should throw if quantity is missing', () => {
+    expect(() => discountCodeGenerator({ length: 23 }, data))
+      .toThrow(/The generator requires valid parameters/)
+  })
+
+  test('should generate codes according to required specifications', () => {
+    const codes = discountCodeGenerator(options, data)
+    expect(Array.isArray(codes)).toBeTruthy()
+    expect(codes.length).toBe(100)
+
+    codes.forEach((codeObject) => {
+      expect(codeObject).toMatchObject(data)
+      expect(codeObject.code).toBeTruthy()
```

Comment:
The [next assertion](https://github.com/commercetools/nodejs/pull/147/files/e742d5c7fbfa08338c92874e73618914e28ca829#diff-c8f4a8fbd9412cfe0adf807a27b5a09bR54) tests the default values.